[AOTI] move model runner into a library

sijiac opened the pull request:
Summary: So that we can import it in fbcode and do some AOTI run in py env

Test Plan: existed AOTI tests

Reviewed By: chenyang78

Differential Revision: D51780021




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

desertfire commented on test/inductor/test_aot_inductor.py:
```diff
@@ -53,114 +51,16 @@
     except ImportError:
         from test_torchinductor import copy_tests, requires_multigpu, TestFailure
+
+    try:
```
Nit: you can merge this with Line 50-53.

sijiac commented:
@pytorchbot merge

sijiac committed:
[AOTI] move model runner into a library (#115220)

Summary:

So that we can import it in fbcode and do some AOTI run in py env

Test Plan: existed AOTI tests

Reviewed By: muchulee8, desertfire, chenyang78

Differential Revision: D51780021

sijiac commented:
@pytorchbot merge
